<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>375</width>
    <height>382</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>FFB Joystick Settings</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <item>
     <widget class="QTabWidget" name="tabWidget_3">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QCheckBox" name="enableConstantViaSinCheckBox">
          <property name="toolTip">
           <string>Some games use sine effect to emulate constant force (for example Raceroom) Check this box for correct effects in such games.</string>
          </property>
          <property name="text">
           <string>Enable constant via sine</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="updateDurationOnEffectReportCheckBox">
          <property name="toolTip">
           <string>Some games do not update effects duration correctly. Keep this box checked unless your game does not stop effects correctly.</string>
          </property>
          <property name="text">
           <string>Update duration on effect report</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="invertDirection_1">
          <property name="toolTip">
           <string>Uncheck/check this box if your wheel turns to wrong side in game</string>
          </property>
          <property name="text">
           <string>Invert direction</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="3" column="0">
           <widget class="QLabel" name="label_21">
            <property name="toolTip">
             <string>Filter for torque output smoothness. Increase value if FFB effects are noisy in game.</string>
            </property>
            <property name="text">
             <string>Torque smoothing</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QSpinBox" name="steeringSmoothingSpinner_1"/>
          </item>
          <item row="0" column="2">
           <widget class="QSpinBox" name="steeringRangeSpinner_1">
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QSpinBox" name="torqueGainSpinner_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QSpinBox" name="springSpinner_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QSlider" name="springSlider_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_25">
            <property name="toolTip">
             <string>Total effect power. Set ot negative values if you need to inver effect direction</string>
            </property>
            <property name="text">
             <string>Torque gain</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_26">
            <property name="toolTip">
             <string>Constant effect power. Set ot negative values if you need to inver effect direction</string>
            </property>
            <property name="text">
             <string>Constant</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_18">
            <property name="toolTip">
             <string>Periodic effect power. Set ot negative values if you need to inver effect direction</string>
            </property>
            <property name="text">
             <string>Periodic</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_23">
            <property name="toolTip">
             <string>Total range of wheel rotation in deegrees</string>
            </property>
            <property name="text">
             <string>Steering range</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSlider" name="steeringSmoothingSlider_1">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSlider" name="steeringRangeSlider_1">
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSlider" name="constantSlider_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QSpinBox" name="constantSpinner_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSlider" name="periodicSlider_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_28">
            <property name="toolTip">
             <string>Spring effect power. Set ot negative values if you need to inver effect direction</string>
            </property>
            <property name="text">
             <string>Spring</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSlider" name="torqueGainSlider_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_27">
            <property name="toolTip">
             <string>Filter for steering output smoothness. Increase value if wheel steering is noisy in game.</string>
            </property>
            <property name="text">
             <string>Steering smoothing</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSlider" name="torqueSmoothingSlider_1">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QSpinBox" name="periodicSpinner_1">
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QSpinBox" name="torqueSmoothingSpinner_1"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Advanced</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QCheckBox" name="checkBoenableSystemSettingsCheckBox">
          <property name="text">
           <string>Enable advanced settings</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout">
          <property name="leftMargin">
           <number>5</number>
          </property>
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="rightMargin">
           <number>5</number>
          </property>
          <property name="bottomMargin">
           <number>5</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Encoder CPR</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="encoderSpinBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
            <property name="maximum">
             <number>99999</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="mechanicalCenterSpinBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="maximum">
             <number>99999</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Mechanical center</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="mechanicalCenterSetButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Set mechanical center to current position</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="calibrateElectricalCenterButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Calibrate electrical poles curve</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="readButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Read from controller</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="writeButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Wright to controller</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>periodicSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>periodicSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>torqueGainSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>torqueGainSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>springSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>springSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>466</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>torqueSmoothingSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>torqueSmoothingSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steeringRangeSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>steeringRangeSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>periodicSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>periodicSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steeringRangeSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>steeringRangeSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoenableSystemSettingsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>calibrateElectricalCenterButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steeringSmoothingSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>steeringSmoothingSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>torqueSmoothingSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>torqueSmoothingSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>constantSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>constantSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>torqueGainSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>torqueGainSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>springSpinner_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>springSlider_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>351</x>
     <y>466</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>467</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>constantSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>constantSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>steeringSmoothingSlider_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>steeringSmoothingSpinner_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoenableSystemSettingsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>encoderSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>121</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoenableSystemSettingsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>mechanicalCenterSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoenableSystemSettingsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>mechanicalCenterSetButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>36</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>134</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoenableSystemSettingsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>mechanicalCenterSetButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
